src/frequenz/sdk/microgrid/_data_sourcing/microgrid_api_source.py

```diff
@@ -461,7 +461,8 @@ async def _update_streams(
             self.comp_data_tasks[comp_id].cancel()
 
         self.comp_data_tasks[comp_id] = asyncio.create_task(
-            run_forever(lambda: self._handle_data_stream(comp_id, category))
+            run_forever(lambda: self._handle_data_stream(comp_id, category)),
+            name=f"run_forever:cid={comp_id}",
```

Comment:
This looks confusing. We have a `run_forever` function. Why not something like:
```suggestion
            name=f"{type(self).__name__}._update_stream({comp_id=}, {category.name})",
```